libpod/container.go

```diff
@@ -1173,6 +1173,54 @@ func (c *Container) Networks() ([]string, bool, error) {
 	return c.networks()
 }
 
+// NetworkMode gets the configured network mode for the container.
+// Get actual value from the database
+func (c *Container) NetworkMode(ctrSpec *spec.Spec) (string, error) {
```

Comment:
Why does this accept a spec? The container struct already contains a spec, and nothing you're accessing here requires something that would be overwritten from the initial user-provided spec. Just using `c.config.Spec` is simpler and safer.

Reply:
@mheon Following function is also going to be used by `generateInspectContainerHostConfig` in inspect code chunk. I think for both `inspect` and `ps --filter` their exists a state where config does not exist **(e.g., because the container was never started)** for such cases I think we can read data from db.

However for inspect block `ctrSpec` is already fetched from state so i thought would be more optimal to reuse the `ctrSpec` as argument inoder to reduce lookup cost.

Reply:
`c.config.Spec` is always valid; there is never a point at which it is not populated. It is not the final spec that is used for the container (we use it to generate the final spec), but it is very close - the changes made to produce a final spec are minimal. As such it is safe to use here - no part of what you are accessing will change during final spec generation.